odl/discr/partition.py

```diff
@@ -982,15 +982,17 @@ def uniform_partition_fromgrid(grid, min_pt=None, max_pt=None):
         min_pt = np.atleast_1d(min_pt)
         min_pt = {i: float(v) for i, v in enumerate(min_pt)}
     else:
-        min_pt.update({i: None for i in range(grid.ndim) if i not in min_pt})
+        min_pt.update({i: None for i in range(grid.ndim)
+                       if i not in min_pt and i - grid.ndim not in min_pt})
```

Comment:
Well the comment farther above explains what happens, clarified it a bit now. Basically the new code does what the old comment says, in contrast to the old code :-)